File: ide/ui/ide-problems/src/main/resources/META-INF/dirigible/ide-problems/view/problems.js
/*
 * Copyright (c) 2022 SAP SE or an SAP affiliate company and Eclipse Dirigible contributors
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-FileCopyrightText: 2022 SAP SE or an SAP affiliate company and Eclipse Dirigible contributors
 * SPDX-License-Identifier: EPL-2.0
 */
angular.module('problems', [])
    .factory('$messageHub', [function () {
        let messageHub = new FramesMessageHub();
        let message = function (evtName, data) {
            messageHub.post({ data: data }, evtName);
        };
        let on = function (topic, callback) {
            messageHub.subscribe(callback, topic);
        };
        return {
            message: message,
            on: on
        };
    }])
    .controller('ProblemsController', ['$scope', '$http', '$messageHub', function ($scope, $http, $messageHub) {
        $scope.selectAll = false;
        $scope.searchText = "";
        $scope.problemsList = [];
        $scope.limit = 25;
        $scope.openedProblem = {};

        function fetchData() {
            $http.get('/services/v4/ops/problems/search', { params: { 'condition': $scope.searchText, 'limit': $scope.limit } }).then(function (response) {
                $scope.problemsList = response.data.result;
                $scope.selectedRows = response.data.selectedRows;
                $scope.totalRows = response.data.totalRows;
            });
            $scope.selectAll = false;
        }

        function refreshList() {
            $scope.searchText = "";
            $scope.limit = 25;

            fetchData();
        }

        refreshList();

        function filterSelectedIds() {
            let selectedIds = [];
            $scope.problemsList.filter(
                function (problem) {
                    if (problem.checked) {
                        selectedIds.push(problem.id)
                    }
                }
            );
            return selectedIds;
        }

        this.refresh = function () {
            refreshList();
        }

        this.nextBatch = function (size) {
            $scope.limit = $scope.limit + size;
            fetchData();
        }

        $scope.checkAll = function () {
            angular.forEach($scope.problemsList, function (problem) {
                problem.checked = $scope.selectAll;
            });
        };

        $scope.updateStatus = function (status) {
            $http.post('/services/v4/ops/problems/update/' + status, filterSelectedIds()).then(function () {
                fetchData();
                $scope.selectAll = false;
            });
        };

        $scope.deleteByStatus = function (status) {
            $http.delete('/services/v4/ops/problems/delete/' + status).then(function () {
                fetchData();
                $scope.selectAll = false;
            });
        }

        $scope.deleteSelected = function () {
            $http.post('/services/v4/ops/problems/delete/selected', filterSelectedIds()).then(function () {
                fetchData();
                $scope.selectAll = false;
            });
        }

        $scope.clear = function () {
            $http.delete('/services/v4/ops/problems/clear').then(function () {
                fetchData();
                $scope.selectAll = false;
            });
        }

        $scope.openFile = function (fullPath) {
            const fullName = fullPath.split("/").pop();
            const [name, extension] = fullName.split('.');

            let msg = {
                "file": {
                    "name": name,
                    "path": "/workspace" + fullPath,
                    "type": "file",
                    "contentType": extension,
                    "label": fullName
                }
            };
            $messageHub.message('ide-core.openEditor', msg);
        }

        $scope.showInfo = function (problem) {
            $scope.openedProblem = problem;
        }

        $('#problemContent').on('hidden.bs.modal', function () {
            $scope.openedProblem = {};
        });
    }]);

File: components/ide/ide-problems/src/main/java/org/eclipse/dirigible/components/ide/problems/domain/Problems.java
/*
 * Copyright (c) 2023 SAP SE or an SAP affiliate company and Eclipse Dirigible contributors
 *
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v2.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v20.html
 *
 * SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and Eclipse Dirigible contributors
 * SPDX-License-Identifier: EPL-2.0
 */
package org.eclipse.dirigible.components.ide.problems.domain;

import java.util.List;

/**
 * The Class Problems.
 */
public class Problems {

    /** The result. */
    private List<Problem> result;
    
    /** The selected rows. */
    private int selectedRows;
    
    /** The total rows. */
    private int totalRows;

    /**
     * Instantiates a new response model.
     */
    public Problems() {
    }

    /**
     * Instantiates a new response model.
     *
     * @param result the result
     * @param selectedRows the selected rows
     * @param totalRows the total rows
     */
    public Problems(List<Problem> result, int selectedRows, int totalRows) {
        this.result = result;
        this.selectedRows = selectedRows;
        this.totalRows = totalRows;
    }

    /**
     * Gets the result.
     *
     * @return the result
     */
    public List<Problem> getResult() {
        return result;
    }

    /**
     * Sets the result.
     *
     * @param result the new result
     */
    public void setResult(List<Problem> result) {
        this.result = result;
    }

    /**
     * Gets the selected rows.
     *
     * @return the selected rows
     */
    public int getSelectedRows() {
        return selectedRows;
    }

    /**
     * Sets the selected rows.
     *
     * @param selectedRows the new selected rows
     */
    public void setSelectedRows(int selectedRows) {
        this.selectedRows = selectedRows;
    }

    /**
     * Gets the total rows.
     *
     * @return the total rows
     */
    public int getTotalRows() {
        return totalRows;
    }

    /**
     * Sets the total rows.
     *
     * @param totalRows the new total rows
     */
    public void setTotalRows(int totalRows) {
        this.totalRows = totalRows;
    }
}
